plugin/content-script.js

```diff
@@ -749,22 +742,13 @@ function onUrlChange() {
 		waitForElement("[class^='ButtonHoverWrapper']", 10000)
 		.then( () => {
 			createOffersHelperContainer();
-
-			watching4histories = setInterval(function() {
-				waitForElement(".HistoryItemWrapper-sc-13gqei4-0", 1000)
-				.then( () => {
-					let histories = Array.prototype.slice.call(document.getElementsByClassName("HistoryItemWrapper-sc-13gqei4-0"));
-					let string = array_to_string(histories);
-					if (string != lastHistory){
-						lastHistory = string;
-						clean_chart(chart);
-						clean_chart(chart3);
-						clean_chart(chart4);
-						updateHistory(histories);
-					}
-				}, () => {} );
-			}, 1000);
-
+				watching4histories = setInterval(function() {
+					waitForElement(".HistoryItemWrapper-sc-13gqei4-0", 1000)
+					.then( () => {
+						if (!firstUpdateHistAttemptOccurred) {updateHistoryWithApiData()}
+						firstUpdateHistAttemptOccurred = true;
+					}, () => {} );
+				}, 250);
```

Comment:
why 250 ?

Reply:
Oops, I forgot to remove this. I was experimenting with different ways to ensure that the async api calls were made without spamming the API endpoint.